buildSrc/src/main/groovy/io/spring/gradle/RemoteProjectTask.groovy

```diff
@@ -0,0 +1,76 @@
+package io.spring.gradle
+
+import groovy.transform.CompileStatic
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.gradle.api.DefaultTask
+import org.gradle.api.Task
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Checkout a project template from a git repository.
+ */
+@CompileStatic
+abstract class RemoteProjectTask extends DefaultTask {
+
+    /**
+     * URI of the git repository.
+     *
+     * Either the remote git repository URL, the path to a local bare git repository or the path to a local git clone.
+     */
+    @Input
+    abstract Property<String> getRemoteUri()
+
+    /**
+     * Git reference to use.
+     */
+    @Input
+    abstract Property<String> getRef()
+
+    /**
+     * Directory where the project template should be copied.
+     */
+    @OutputDirectory
+    final File outputDirectory = project.file("$project.buildDir/$name")
+
+    @TaskAction
+    void checkoutAndCopy() {
+        outputDirectory.deleteDir()
+        File checkoutDir = checkout(this, remoteUri.get(), ref.get())
+        moveToOutputDir(checkoutDir, outputDirectory)
+    }
+
+    private static File cleanTemporaryDir(Task task, File tmpDir) {
+        if (tmpDir.exists()) {
+            task.project.delete(tmpDir)
+        }
+        return tmpDir
+    }
+
+    static File checkout(Task task, String remoteUri, String ref) {
+        checkout(task, remoteUri, ref, task.getTemporaryDir())
+    }
+
+    @TypeChecked(TypeCheckingMode.SKIP)
+    static File checkout(Task task, String remoteUri, String ref, File checkoutDir) {
+        cleanTemporaryDir(task, checkoutDir)
+        task.project.exec {
+            commandLine = ["git", "clone", "--no-checkout", remoteUri, checkoutDir.absolutePath]
+            errorOutput = System.out
```

Comment:
Is there a reason you are using `System.out` instead of the default `System.err`? 

Also, I'm wondering if you've considered integrating with the logger.

Reply:
No reason, I've just used the wrong `PrintStream`. I'll take a look at the logger